ui/src/app/pages/data/cohort-review/cohort-review.tsx

```diff
@@ -20,81 +26,111 @@ const styles = reactStyles({
+      this.loadCohort();
+    }
 
-  componentDidMount(): void {
-    this.props.hideSpinner();
-    const {ns, wsid, cid} = urlParamsStore.getValue();
-    const {accessLevel, cdrVersionId} = currentWorkspaceStore.getValue();
-    this.setState({readonly: accessLevel === WorkspaceAccessLevel.READER});
-    cohortReviewApi().getParticipantCohortStatuses(ns, wsid, cid, +cdrVersionId, {
-      page: 0,
-      pageSize: 25,
-      sortOrder: SortOrder.Asc,
-      filters: {items: []}
-    }).then(resp => {
-      const {cohortReview, queryResultSize} = resp;
-      currentCohortReviewStore.next(cohortReview);
-      queryResultSizeStore.next(queryResultSize);
-      const reviewPresent = cohortReview.reviewStatus !== ReviewStatus.NONE;
-      this.setState({reviewPresent});
-      if (reviewPresent) {
-        navigate(['workspaces', ns, wsid, 'data', 'cohorts', cid, 'review', 'participants']);
+    loadCohort() {
+      const {ns, wsid, cid} = this.props.urlParams;
+      const {accessLevel, cdrVersionId} = this.props.workspace;
+      this.setState({readonly: accessLevel === WorkspaceAccessLevel.READER});
+
+      if (!cid) {
+        return;
       }
-    });
-    cohortsApi().getCohort(ns, wsid, cid).then(cohort => this.setState({cohort}));
-    if (!visitsFilterOptions.getValue()) {
-      cohortBuilderApi().findCriteriaBy(
-        ns, wsid, Domain[Domain.VISIT], CriteriaType[CriteriaType.VISIT]
-      ).then(response => {
-        visitsFilterOptions.next([
-          {value: null, label: 'Any'},
-          ...response.items.map(option => {
-            return {value: option.name, label: option.name};
-          })
-        ]);
+
+      cohortReviewApi().getParticipantCohortStatuses(ns, wsid, cid, +cdrVersionId, {
+        page: 0,
+        pageSize: 25,
+        sortOrder: SortOrder.Asc,
+        filters: {items: []}
+      }).then(resp => {
+        const {cohortReview, queryResultSize} = resp;
+        currentCohortReviewStore.next(cohortReview);
+        queryResultSizeStore.next(queryResultSize);
+        const reviewPresent = cohortReview.reviewStatus !== ReviewStatus.NONE;
+        this.setState({reviewPresent});
+        if (reviewPresent) {
+          this.props.navigate(['workspaces', ns, wsid, 'data', 'cohorts', cid, 'review', 'participants']);
+        }
       });
+      cohortsApi().getCohort(ns, wsid, cid).then(cohort => this.setState({cohort}));
+      if (!visitsFilterOptions.getValue()) {
+        cohortBuilderApi().findCriteriaBy(
+          ns, wsid, Domain[Domain.VISIT], CriteriaType[CriteriaType.VISIT]
+        ).then(response => {
+          visitsFilterOptions.next([
+            {value: null, label: 'Any'},
+            ...response.items.map(option => {
+              return {value: option.name, label: option.name};
+            })
+          ]);
+        });
+      }
     }
-  }
 
-  reviewCreated = () => {
-    this.setState({reviewPresent: true});
-  }
+    componentDidUpdate(prevProps: Readonly<Props>) {
```

Comment:
Do you think we should create tickets to turn all the components where we've had to use this pattern into function components?